src/data_types/tensor.hpp

```diff
@@ -6,60 +6,68 @@
 #include <ddc/ddc.hpp>
 
 #include "ddc_aliases.hpp"
+#include "tensor_common.hpp"
 #include "tensor_index_tools.hpp"
 #include "vector_index_tools.hpp"
 
+namespace detail {
+template <class ElementType, class... ValidIndexSet>
```

Comment:
Shall we just template by the number of elements ? That would avoid variadic templates and reuse existing classes for different ValidIndexSet of the same size.

Reply:
I don't see any way to preserve the functionalities for which this class was designed while doing that.

For example:
```cpp
DTensor<VectorIndexSpace<R, Theta>, VectorIndexSpace<X,Y>> Jacobian;
```
would become:
```cpp
DTensor<2> Jacobian;
```
?
At this point you may as well use a normal Kokkos array/std::array you would lose all static typing advantages

Reply:
Done [8a53652](https://github.com/gyselax/gyselalibxx/pull/262/commits/8a536527823a7210f4fac43d04d615cd7483f23a)